ha/util/conf_store.py

```diff
@@ -221,3 +221,30 @@ def get_disk_list_for_cvg(index, node_id, cvg_id):
         except Exception as e:
             Log.error(f"Unable to fetch Disk list from GConf. Error {e}")
             raise Exception(f"Unable to fetch Disk list. Error {e}")
+
+    @staticmethod
+    def get_control_pods(index):
+        """
+        Returns machine_id for control-node pod.
+        Args:
+            index(str): index of conf file
+            parent key : node
+            search key: type
+            search val: control_node
+        Returns:
+            list: list of node_ids
+
+        >>> Conf.search("cortx", 'node', 'type', 'control_node')
+        ['node>a3710eee36644f2fb60ba19486664495>type']
+        """
+        node_id = []
+        try:
+            node_key = Conf.search(index, GconfKeys.NODE_CONST.value,
```

Comment:
Please check if this will work. 
Refer API spec for Conf.search: https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/316342315/Conf+Store#Search-API

It should be in this format: 
Conf.search('index', 'parent_key', 'key_to_search', 'value_to_search')
Where value_to_search is optional and should be used only if you are looking for a specific key with a specific value!

Reply:
Hi @rahul-tripathi-git 
Conf.search('index', 'parent_key', 'key_to_search', 'value_to_search')
This is the exact format we are using:
**Conf.search('test', 'node', 'type', 'control_node')**
index -> test
parent key -> node
key to search -> type
value_to_search -> control_node

Do you require any changes in the PR?

Reply:
Ok @gargpalak048 
We are good. resolving this comment.

Reply:
@rahul-tripathi-git , 'value_to_search' needs to come from https://github.com/Seagate/cortx-utils/blob/main/py-utils/src/utils/cortx/const.py or similar constant  correct ? 
Otherwise the code will break if the value  "control_node" is changed in future.

Is this available as constant anywhere ? or it is okay to use "control_node" directly ?

cc @shaileshsvc @gargpalak048

Reply:
I see that this is specific to HA. and it is already put in a similar const.py file.

Not sure if this level of key/value is maintained somewhere. For the [const.py in utils](https://github.com/Seagate/cortx-utils/blob/main/py-utils/src/utils/cortx/const.py), it was added by cortx-prvsnr to have these services in a common place.

@ujjwalpl , can you please comment on the query above?

Reply:
node types can be anything and we need to avoid hard coding. Way to get it is.. 
key = Conf.search(index, 'node', 'components', 'csm')

output key will contain node which has csm component in format 'node>node_id>components..'
Extract node_id and prepare a key as 
node_type = 'node>$node_id>name'  <<< This will return control node.

Reply:
ACK.
thanks @ujjwalpl.

Now to fetch control node machine-id, 
using below query

> Conf.search(index, 'node', name, 'csm')
and fetching node_id from there.

Updated the patch with above changes.
Cc: @ArchanaLimaye, @rahul-tripathi-git

Reply:
Looks good Palak!
`>>>key = Conf.search('index', 'node', 'name', 'csm')`
`>>>print( key)`
`['node>9995f539f4f770e2a3fe9e2e615c32a8>components[1]>name']`
`>>>node_type = Conf.get('i', "%s>name" %key[0].rsplit(">", 2)[0])`
`>>>print(node_type)`
`'control-node'`